src/Control/Effect/Diagnostics.hs

```diff
@@ -42,9 +49,24 @@ import Prelude
 
 data Diagnostics m k where
   Fatal :: ToDiagnostic diag => diag -> Diagnostics m a
-  Recover :: m a -> Diagnostics m (Maybe a)
+  Recover' :: m a -> Diagnostics m (Either SomeDiagnostic a)
```

Comment:
The prime is meant to match the definition of `recover'` below; we're keeping the previous `recover`, which is just a thin wrapper around `recover'`

The naming convention I've been using is to add a `'` to denote the "more-complicated" variants of functions

Reply:
TLDR: it's up to you, I'm totally fine either way.

> The naming convention I've been using is to add a ' to denote the "more-complicated" variants of functions

If that's the convention we're using, then I think we should usually remove prime marks unless more than one variant exists, or there's an exceptional reason.

I'm actually fine with either keeping or removing this particular prime, since it's not really necessary on its own, but it does match the effect function.  My preference is to remove it, since there's no real need to connect the names, and primes without variants add noise, but I'm probably not going to remember it's even there once this merges.